src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -555,7 +557,22 @@ public void AddLowercase(CultureInfo culture)
                     }
                     else
                     {
-                        AddLowercaseRange(range.First, range.Last);
+                        char lower = culture.TextInfo.ToLower(range.First);
+                        char upper = culture.TextInfo.ToLower(range.Last);
+                        if (range.Last - range.First == upper - lower)
+                        {
+                            AddLowercaseRange(range.First, range.Last);
+                        }
+                        else
+                        {
+                            // Bug fix: Unicode `Symbol`s sometimes exist in the middle of character ranges. char.ToLower(Symbol) returns Symbol. In these cases, we cannot use an offset to find the lowercase chars. For ex: https://github.com/dotnet/runtime/issues/36149
```

Comment:
(soapbox).. 

I once read someone's description of the ideal state of a mature codebase as one where the code appeared to have been written by a single highly productive perfect person in one sitting. The idea was the code is fully self consistent and it reads easily and intuitively. In that philosophy, a bug fix is working to bring it closer to that ideal state. History matters, but that is what git blame and github search exists for. Historical commentary in a codebase often doesn't age well -- it can be irrelevant, become outdated (eg dead links), and looks scary to refactor.

I do think it's great to have comments like this where it's not obvious why the code is written a certain way - since that hypothetical perfect person would have written comments of that kind.

Tests to me are different - they're inevitably an idiosyncratic collection of independent chunks of code for particular bug fixes, etc and links and bug ID's there make more sense.

I know this is partly a matter of taste and @stephentoub has a slightly different preference, but clearly we're just picking places on a dial, since we make many bug fixes without including Github URL's

Reply:
I agree about the bug fix part; I feel the same way about tests.

Reply:
Well it's good that the whole else block (containing the comment and URL) is going away then :D

Reply:
> tests labeled "regression" really bug me :-)

That's interesting, and I shall try to avoid them. Let's make our tests look like they were written by perfect people too!